scss/_patterns_navigation.scss

```diff
@@ -210,81 +213,108 @@ $nav-border-bottom-thickness: 1px;
   }
 
   // Theming
+  // Manually overriding the theming:
+  $color-navigation-background: null !default;
+  $color-navigation-text: null !default;
+  $lightness-threshold: 70;
+
   @if (map-get($theme-default--dark, nav) == true) {
+    // dark theme
+    $color-navigation-background: $colors--dark-theme__background-highlighted !default;
+    $color-navigation-text: if(lightness($color-navigation-background) > $lightness-threshold, $colors--light-theme__text-default, $colors--dark-theme__text-default) !default;
+
     .p-navigation {
-      @extend %p-navigation--dark-theme;
+      @include p-navigation--dark-theme($color-navigation-background, $color-navigation-text, false);
     }
 
     .p-navigation.is-light {
-      @extend %p-navigation--light-theme;
+      @include p-navigation--light-theme($color-navigation-background, $color-navigation-text, true);
     }
   } @else {
+    // light theme
+    $color-navigation-background: $colors--light-theme__background !default;
+    $color-navigation-text: if(lightness($color-navigation-background) > $lightness-threshold, $colors--light-theme__text-default, $colors--dark-theme__text-default) !default;
+
     .p-navigation {
-      @extend %p-navigation--light-theme;
+      @include p-navigation--light-theme($color-navigation-background, $color-navigation-text, false);
     }
 
     .p-navigation.is-dark {
-      @extend %p-navigation--dark-theme;
+      @include p-navigation--dark-theme($color-navigation-background, $color-navigation-text, true);
     }
   }
 
-  %p-navigation--dark-theme {
-    background-color: map-get($colors--dark-theme, background-highlighted);
+  @debug lightness($color-navigation-background);
+}
 
-    .p-navigation__link > a,
-    .p-navigation__toggle--close,
-    .p-navigation__toggle--open {
-      &,
-      &:visited,
-      &:focus {
-        color: map-get($colors--dark-theme, text-default);
-      }
+@mixin p-navigation--light-theme($color-navigation-background, $color-navigation-text, $override-default: false) {
+  @if ($override-default) {
```

Comment:
So if override is true we hard set the colors to light instead of using the colors passed to the mixin?

Reply:
Override: true is only needed when we invoke the mixin from within .is-dark is-light classes. From the perspective of a vanilla user, If you add is-dark to your markup, you get the stock dark theme.

By defsult the mixin uses $color-navigation-background, if set.